From 441b3fd88149c03c0eb4ba373f3c6633df2f3196 Mon Sep 17 00:00:00 2001
From: Bhupesh Sharma <bhsharma@redhat.com>
Date: Fri, 29 Jun 2018 16:14:48 +0530
Subject: [PATCH] arm64: Fix calculation of page_offset in case we are running
 cases other than mem-usage

Patch f49ca13e5eed5bbdc69e0fd5ef099cb46050cb3d added '--mem-usage'
support for arm64 architecture.

However, we also need to make sure that the calculation of
'page_offset' is valid in case we are running cases other than
'--mem-usage'.

This patch does the same. I tested this patch on my qualcomm and apm
mustang arm64 boards.

Signed-off-by: Bhupesh Sharma <bhsharma@redhat.com>
---
 arch/arm64.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/makedumpfile-1.6.2/arch/arm64.c b/makedumpfile-1.6.2/arch/arm64.c
index c9dab677f2c9..2fd3e1874376 100644
--- a/makedumpfile-1.6.2/arch/arm64.c
+++ b/makedumpfile-1.6.2/arch/arm64.c
@@ -222,6 +222,12 @@ get_stext_symbol(void)
 int
 get_machdep_info_arm64(void)
 {
+	/* Check if va_bits is still not initialized. If still 0, call
+	 * get_versiondep_info() to initialize the same.
+	 */
+	if (!va_bits)
+		get_versiondep_info_arm64();
+
 	if (!calculate_plat_config()) {
 		ERRMSG("Can't determine platform config values\n");
 		return FALSE;
-- 
2.7.4

